src/iter/index.ts

```diff
@@ -45,19 +45,27 @@ export function iter <M extends QueryMethodTypes> (iterResult: IterResult<M>, op
       for (let j = 0; j < poslist.length; j++) {
         const res = poslist[j]
         if (until && res > until) {
-          return emitResult(iterResult)
+          return
         }
 
         if (res >= dtstart) {
           const rezonedDate = rezoneIfNeeded(res, options)
-          if (!iterResult.accept(rezonedDate)) {
-            return emitResult(iterResult)
+          if (iterResult.accept(rezonedDate)) {
+            yield rezonedDate
```

Comment:
This is the main change, only yield data when the iterResult accepts it and pause execution until the next iteration.